tests/integration/pipeline/base.py

```diff
@@ -0,0 +1,119 @@
+import os
+import re
+import shutil
+from pathlib import Path
+from typing import List
+from unittest import TestCase
+
+import boto3
+
+from tests.testing_utils import run_command_with_input
+
+
+class PipelineBase(TestCase):
+    def base_command(self):
+        command = "sam"
+        if os.getenv("SAM_CLI_DEV"):
+            command = "samdev"
+
+        return command
+
+    def run_command_with_inputs(self, command_list, inputs: List[str]):
+        return run_command_with_input(command_list, ("\n".join(inputs) + "\n").encode())
+
+
+class InitIntegBase(PipelineBase):
+    generated_files: List[Path] = []
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        # we need to compare the whole generated template, which is
+        # larger than normal diff size limit
+        cls.maxDiff = None
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.generated_files = []
+
+    def tearDown(self) -> None:
+        for generated_file in self.generated_files:
+            if generated_file.is_dir():
+                shutil.rmtree(generated_file, ignore_errors=True)
+            elif generated_file.exists():
+                generated_file.unlink()
+        super().tearDown()
+
+    def get_init_command_list(
+        self,
+    ):
+        command_list = [self.base_command(), "pipeline", "init"]
+        return command_list
+
+
+class BootstrapIntegBase(PipelineBase):
+    stack_names: List[str]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.cf_client = boto3.client("cloudformation")
+        pass
+
+    def setUp(self):
+        self.stack_names = []
+        super().setUp()
+
+    def tearDown(self):
+        if self.stack_names:
+            for stack_name in self.stack_names:
+                self.cf_client.delete_stack(StackName=stack_name)
+        shutil.rmtree(os.path.join(os.getcwd(), ".aws-sam", "pipeline"), ignore_errors=True)
```

Comment:
This line is dangerous; what if I ran `sam pipeline bootstrap` (for real use-case not for integ-tests) then later on I ran the integration tests, I will lose the real data that I don't want to lose.

Also, if we later on added additional files to `.aws-sam/pipeline` folder it will be lost as well.

Reply:
It is our practice to do that, see tests/integration/deploy/test_deploy_command.py:56

Also it is unlikely to run `sam pipeline bootstrap` and integration test in the same directory.

Reply:
yeah, I agree it is unlikely. So, this should be fine.